lib/commons/aria/get-role.js

```diff
@@ -32,14 +113,26 @@ function getRole(node, { noImplicit, fallback, abstracts, dpub } = {}) {
 		return isValidRole(role, { allowAbstract: abstracts });
 	});
 
-	const explicitRole = validRoles[0];
+	const explicitRole = validRoles[0] || null;
+	if (noImplicit) {
+		return explicitRole;
+	}
 
 	// Get the implicit role, if permitted
-	if (!explicitRole && !noImplicit) {
-		return getImplicitRole(node);
+	const implicitRole = resolveImplicitRole(vNode);
+
+	// role conflict resolution
+	// Source: https://www.w3.org/TR/wai-aria-1.1/#conflict_resolution_presentation_none
+	// See also: https://github.com/w3c/aria/issues/1270
+	if (
+		!explicitRole ||
+		(['presentation', 'none'].includes(explicitRole) &&
+			(hasGlobalAriaAttributes(vNode) || isFocusable(node)))
+	) {
+		return implicitRole;
```

Comment:
So, here's a problem.... Chrome and Safari don't return the implicit role for certain elements. For example:

```html
<ul role="none" aria-label> <li>hello</li> </ul>
```

This is returned as "generic". Only Firefox actually returns this as a list. Buttons and links do get their role back. This seems to be exclusively based on the nodeName. We're going to need to look into what we want to do here.